### PR TITLE
R4 CarePlan Read and Search Documentation Added 

### PR DESCRIPTION
**Pull request by aki-mathur**
Description
 The R4 documentation for CarePlan Read and Search

 pre-merge validation:

CareProvision
CarePlanView
CarePlanSearchParams

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/care-provision/care-plan.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+### Parameters
+
+ Name         | Required? | Type          | Description
+--------------|-----------|---------------|--------------
+ `_id`        | See notes | [`token`]     | The logical resource id associated with the resource.
+ `date`       | N         | [`date`]      | Time period plan covers. Example: `date=ge2016&date=le2017 (Jan 1, 2016 - Dec 31, 2017)` or `date=ge2016-08-24T12:00:00.000Z&date=le2017-01-24T12:00:00.000Z`
+ `patient`    | See notes | [`reference`] | Who care plan is for. Example: `patient=1316024`
+ `category`   | See notes | [`token`]     | Only support 'assess-plan'
+ [`_count`]   | N         | [`number`]    | Number of results per page.
+ [`_revinclude`]| N       |               | This can be a future enhancement and will not hold up the initial deployment of CarePlan. Required for USCDI
```
Comment (sonikrish) [inline content/millennium/r4/care-provision/care-plan.md:63]: This you can remove for now

File: lib/resources/r4/care_plan.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+name: CarePlan
+field_name_base_url: http://hl7.org/fhir/R4/careplan-definitions.html#CarePlan
+fields:
+- name: category
+  required: 'No'
+  type: List of CodeableConcept
+  binding:
+    description: >
+      Identifies what "kind" of plan this is to support differentiation between multiple co-existing plans; e.g. "Home health", "psychiatric", "asthma", "disease management", etc.
```
Comment (sonikrish) [inline lib/resources/r4/care_plan.yaml:10]: Can add this This is also used to return a CareTeam (CarePlan with participants). "assess-plan"

File: content/millennium/r4/care-provision/care-plan.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+The R4 CarePlan resource can be used to represent care plans.
+
+### Care Plans
+
+The CarePlan resource describes the intentions of how one or more practitioners plan to deliver care to a specific patient. Care plans are used in a variety of areas and scopes ranging from encounter or visit specific to cross-visits.  Care plans are used to group activities, goals and/or practitioners to create context. Care plans are intended to be specific to the patient. It is possible for a planned activity to stand alone without a care plan such as a scheduled appointment.  
+
+The following fields are returned if valued:
+
+* [Id](http://hl7.org/fhir/r4/resource-definitions.html#Resource.id){:target="_blank"}
```
Comment (sonikrish) [inline content/millennium/r4/care-provision/care-plan.md:20]: Can change it to CarePlan id

File: content/millennium/r4/care-provision/care-plan.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+* [Meta](http://hl7.org/fhir/r4/careplan-definitions.html#CarePlan.meta){:target="_blank"}
```
Comment (sonikrish) [inline content/millennium/r4/care-provision/care-plan.md:21]: you can remove meta from here

File: content/millennium/r4/care-provision/care-plan.md (hunk 1 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Removed 'revinclude'

File: content/millennium/r4/care-provision/care-plan.md (hunk 3 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Updated it to CarePlan Id

File: content/millennium/r4/care-provision/care-plan.md (hunk 4 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Removed meta and text

File: content/millennium/r4/care-provision/care-plan.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+_Implementation Notes_
+
+* The [CarePlan.activity] element is not supported and will not be returned. This means that the [CarePlan.activity.detail.status] and [CarePlan.activity.detail.prohibited] modifier elements will not be returned.
+
+* The description of the CarePlan is included in the [CarePlan.text.div] element as ASCII text.
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
```
Comment (praneethk2094) [inline content/millennium/r4/care-provision/care-plan.md:50]: Do we support patient access?

File: lib/resources/example_json/r4_examples_care_plan.rb [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
...gens/LSD, Heroin, Inhalants/Glues/Solvents, Marijuana, Methamphetamines, Prescription medications, Daily, Previous treatment: Treatment center, Inpatient, Outpatient. Started age 8 Years. IV drug use: Yes. Drug use interferes with work/home: Yes. Ready to change: No. Household substance abuse concerns: No.&#10;Tobacco&#10;Current every day smoker, Cigars, hand, Household tobacco concerns: Yes&#10;Allergies&#10;oxyCODONE&#10;Peanuts&#10;sulfa drugs&#65533;&#65533;(Anaphylactic reaction)&#10;uranium&#10;iodine&#65533;&#65533;(Hives, Sneezing)&#10;penicillin V potassium&#65533;&#65533;(Hives)&#10;skin test antigens, multiple&#10;Home Medication List&#10;Aceon 2 mg oral tablet, 1 tab(s), Oral, Daily, 11 refills&#10;acetaminophen-tramadol&#10;Physical Exam&#10;Vitals &amp; Measurements&#10;Lab Results&#10;Radiology Results&#10;Assessment/Plan&lt;/p&gt;&lt;p&gt;&lt;b&gt;Status&lt;/b&gt;: Active&lt;/p&gt;&lt;p&gt;&lt;b&gt;Intent&lt;/b&gt;: Plan&lt;/p&gt;&lt;p&gt;&lt;b&gt;Category&lt;/b&gt;: Assessment and Plan of Treatment&lt;/p&gt;&lt;p&gt;&lt;b&gt;Author&lt;/b&gt;: Shaik, Faiz Ahmed&lt;/p&gt;&lt;p&gt;&lt;b&gt;Effective Period&lt;/b&gt;: Mar 10, 2016  7:43 P.M. UTC&lt;/p&gt;&lt;/div&gt;"
```
Comment (praneethk2094) [inline lib/resources/example_json/r4_examples_care_plan.rb:13]: Can we avoid formatting the div converting the # into html entity?

File: content/millennium/r4/care-provision/care-plan.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+## Retrieve by id
+
+List an individual CarePlan by its id:
+
+    GET /CarePlan/:id
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
```
Comment (praneethk2094) [inline content/millennium/r4/care-provision/care-plan.md:112]: Can we update this to reflect that patient authorization is currently not supported.

File: content/millennium/r4/care-provision/care-plan.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+### Parameters
+
+ Name         | Required? | Type          | Description
+--------------|-----------|---------------|--------------
+ `_id`        | See notes | [`token`]     | The logical resource id associated with the resource.
+ `date`       | N         | [`date`]      | Time period plan covers. Example: `date=ge2016&date=le2017 (Jan 1, 2016 - Dec 31, 2017)` or `date=ge2016-08-24T12:00:00.000Z&date=le2017-01-24T12:00:00.000Z`
+ `patient`    | See notes | [`reference`] | Who care plan is for. Example: `patient=1316024`
+ `category`   | See notes | [`token`]     | Only support 'assess-plan'
```
Comment (praneethk2094) [inline content/millennium/r4/care-provision/care-plan.md:59]: Do we want to add description to explain how to utilize this parameter. The note should already cover that we only support assess-plan. Thoughts. 

Checkout DSTU2 for example: https://fhir.cerner.com/millennium/dstu2/care-provision/care-plan/

File: lib/resources/example_json/r4_examples_care_plan.rb [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
```
[the remark this answers is not in the document]
Comment (aki-mathur) [reply lib/resources/example_json/r4_examples_care_plan.rb]: Resolved

File: content/millennium/r4/care-provision/care-plan.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+The CarePlan resource describes the intentions of how one or more practitioners plan to deliver care to a specific patient. Care plans are used in a variety of areas and scopes ranging from encounter or visit specific to cross-visits.  Care plans are used to group activities, goals and/or practitioners to create context. Care plans are intended to be specific to the patient. It is possible for a planned activity to stand alone without a care plan such as a scheduled appointment.  
+
+The following fields are returned if valued:
+
+* [CarePlan Id](http://hl7.org/fhir/r4/resource-definitions.html#Resource.id){:target="_blank"}
+* [Status](http://hl7.org/fhir/r4/careplan-definitions.html#CarePlan.status){:target="_blank"}
+* [Intent (plan)](http://hl7.org/fhir/r4/careplan-definitions.html#CarePlan.intent){:target="_blank"}
+* [Category (assess-plan)](http://hl7.org/fhir/r4/careplan-definitions.html#CarePlan.category){:target="_blank"}
+* [Title](http://hl7.org/fhir/r4/careplan-definitions.html#CarePlan.title){:target="_blank"}
+* [Description](http://hl7.org/fhir/R4/careplan-definitions.html#CarePlan.description) {:target="_blank"}
```
[the remark this answers is not in the document]
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Remove the space, Yes we are supporting description

File: content/millennium/r4/care-provision/care-plan.md (hunk 5 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Removed the patient access

File: lib/resources/example_json/r4_examples_care_plan.rb (hunk 6 above)
Comment (aki-mathur) [reply lib/resources/example_json/r4_examples_care_plan.rb]: Resolved

File: content/millennium/r4/care-provision/care-plan.md (hunk 7 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Removed patient authorization

File: content/millennium/r4/care-provision/care-plan.md (hunk 8 above)
Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: Updated with example

File: content/millennium/r4/care-provision/care-plan.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+# CarePlan
+
+* TOC
+{:toc}
+
+## Overview
+
+The R4 CarePlan resource can be used to represent care plans.
+
+### Care Plans
```
Comment (praneethk2094) [inline content/millennium/r4/care-provision/care-plan.md:14]: I think we can remove this header.

Comment (aki-mathur) [reply content/millennium/r4/care-provision/care-plan.md]: removed the care plans header

**Comment by ebbp123** [comment]
Live Verification:
Screen Shot 2020-07-14 at 4 29 41 PM
Screen Shot 2020-07-14 at 4 30 22 PM
Screen Shot 2020-07-14 at 4 30 44 PM

**Pull request merged**